### PR TITLE
basic lfi emulation

### PR DESCRIPTION
**Pull request by afeena**
Added fstab, group, hosts, mtab, new passwd, shadow and environ files (copied from the real system), added new LFI emulator

### CODE REVIEW

File: data/virtualdocs/linux/etc/group [hunk 1]
```diff
@@ -0,0 +1,69 @@
+root:x:0:
+daemon:x:1:
+bin:x:2:
+sys:x:3:
+adm:x:4:syslog,jpotato
```
Comment (glaslos) [inline data/virtualdocs/linux/etc/group]: jpotato

File: data/virtualdocs/linux/etc/group [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+avahi-autoipd:x:119:
+avahi:x:120:
+bluetooth:x:121:
+scanner:x:122:saned
+colord:x:123:
+pulse:x:124:
+pulse-access:x:125:
+rtkit:x:126:
+saned:x:127:
+jpotato:x:1000:
```
Comment (glaslos) [inline data/virtualdocs/linux/etc/group]: jpotato

File: data/virtualdocs/linux/etc/group [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+sambashare:x:128:jpotato
```
Comment (glaslos) [inline data/virtualdocs/linux/etc/group]: jpotato

File: data/virtualdocs/linux/etc/hosts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+127.0.0.1       localhost
+127.0.1.1       jpotato-ubuntu16
```
Comment (glaslos) [inline data/virtualdocs/linux/etc/hosts]: remove that line too :)

File: data/virtualdocs/linux/etc/passwd [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+dnsmasq:x:112:65534:dnsmasq,,,:/var/lib/misc:/bin/false
+colord:x:113:123:colord colour management daemon,,,:/var/lib/colord:/bin/false
+speech-dispatcher:x:114:29:Speech Dispatcher,,,:/var/run/speech-dispatcher:/bin/false
+hplip:x:115:7:HPLIP system user,,,:/var/run/hplip:/bin/false
+kernoops:x:116:65534:Kernel Oops Tracking Daemon,,,:/:/bin/false
+pulse:x:117:124:PulseAudio daemon,,,:/var/run/pulse:/bin/false
+rtkit:x:118:126:RealtimeKit,,,:/proc:/bin/false
+saned:x:119:127::/var/lib/saned:/bin/false
+usbmux:x:120:46:usbmux daemon,,,:/var/lib/usbmux:/bin/false
+jpotato:x:1000:1000:jpotato,,,:/home/jpotato:/bin/bash
```
Comment (glaslos) [inline data/virtualdocs/linux/etc/passwd]: and this

File: data/virtualdocs/linux/etc/shadow [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+dnsmasq:*:16911:0:99999:7:::
+colord:*:16911:0:99999:7:::
+speech-dispatcher:!:16911:0:99999:7:::
+hplip:*:16911:0:99999:7:::
+kernoops:*:16911:0:99999:7:::
+pulse:*:16911:0:99999:7:::
+rtkit:*:16911:0:99999:7:::
+saned:*:16911:0:99999:7:::
+usbmux:*:16911:0:99999:7:::
+jpotato:$6$Yvu1OcDR$NlzEo34sove.n83cBjrgCt7C/9sqhlVJoXf/InQzKIOa.08KMrGpTXN803J6Szi5MBfuuQyPk9WGMhd1ROwPv/:16960:0:99999:7:::
```
Comment (glaslos) [inline data/virtualdocs/linux/etc/shadow]: here

File: data/virtualdocs/linux/proc/self/environ [hunk 7]
```diff
@@ -0,0 +1,16 @@
+USER=jpotato
+LOGNAME=jpotato
+HOME=/home/jpotato
+PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
+MAIL=/var/mail/jpotato SHELL=/bin/bash
+LANG=en_US.UTF-8
+LC_NUMERIC=ru_RU.UTF-8
```
Comment (glaslos) [inline data/virtualdocs/linux/proc/self/environ]: go for just en_US.UTF-8

File: lfi_emulator.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    def __init__(self, root_path):
+        self.root_path = root_path
+        self.whitelist = []
+
+    def available_files(self):
+        for root, dirs, files in os.walk(os.path.join(self.root_path, 'data/virtualdocs/')):
+            for f in files:
+                self.whitelist.append(os.path.join(root, f))
+
+    def lfi_result(self, file):
```
Comment (glaslos) [inline lfi_emulator.py]: `file` shadows `file`

**Comment by glaslos** [comment]
Can you rebase this branch?

**Comment by afeena** [comment]
rebased

**Pull request merged**